src/main/java/com/google/api/codegen/transformer/py/PythonGapicSurfaceTransformer.java

```diff
@@ -175,6 +189,63 @@ private void addFieldsImports(ModelTypeTable typeTable, Iterable<FieldModel> fie
     }
   }
 
+  /**
+   * Generates a list of standalone sample view models for the the API, at most one for each calling
+   * form for each method, as specified in each method view model's sampleValueSetsModel (which
+   * ultimately derives from user-provided configuration). If no samples are configured, no sample
+   * files are produced.
+   *
+   * <p>TODO(vchudnov-g): Properly set the method form in the views thus generated.
+   *
+   * @param context the interface for whose methods the sample files will be generated
+   * @param apClass the previously generated API class view model for whose methods sample files
+   *     will be generated. The view models returned for each method are modified clones of the view
+   *     models for the methods in apiClass.
+   * @return A list of view models, each one corresponding to a distinct method sample
+   */
+  private Iterable<ViewModel> generateSampleClassesForApi(
+      GapicInterfaceContext context, DynamicLangXApiView apiClass) {
+    SurfaceNamer namer = context.getNamer();
+    ImmutableList.Builder<ViewModel> viewModelBuilder = ImmutableList.builder();
+
+    final Builder sampleClassBuilder = apiClass.toBuilder();
+    sampleClassBuilder.templateFileName(STANDALONE_SAMPLE_TEMPLATE_FILENAME);
+
+    for (ApiMethodView view : apiClass.apiMethods()) {
+      OptionalArrayMethodView methodView = (OptionalArrayMethodView) view;
```

Comment:
I'm uneasy of the general approach here, where the client surface is generated first, and then you crack it back open as a basis for the samples. For all of the other stuff we generate, it all starts from the source data - this is done for the client, in-code samples, and tests. I'd prefer to have your samples work the same way instead of working differently.

Reply:
OK, revamped this in both Python and Java. I'm not completely convinced it's simpler, but the extra structure did lead me to find an error earlier.